lib/Sources/BluetoothEngine/ZombieDetector.swift

```diff
@@ -0,0 +1,57 @@
+import Bluetooth
+import BluetoothClient
+import BluetoothMessage
+import Foundation
+
+/**
+ The recentlyConnectedPeripherals are those that have successfully been connected.
+ It is updated by connect/disconnect events. We use the mis-match between
+ recentlyConnectedPeripherals and the underlying CBPeripheral.state to
+ recognize when the system has failed to propagate disconnection events.
+ Zombies are those with an underlying peripheral state of disconnected where
+ we have not seen a disconnect event.
+
+ We see this happen when BLE is diabled via Control Center - the peripherals
+ silently change to a disconnected state but the delegate never sends a disconnect.
+ This is not the case when BLE is disabled via Settings which does the expected
+ thing and immediately triggers the delegate to send a disconnect event.
+
+ Tracking issue: rdar://17036865
```

Comment:
I'm not familiar with `rdar` links. How does one view this?

Reply:
It is internal to Apple and is not public, so the hyperlink would only be active for Apple developers if they happen to stumble across this. So not exactly useful to us but it is standard practice to refer to Radars using the official URI format.